connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/Email.java

```diff
@@ -0,0 +1,259 @@
+import jakarta.mail.internet.MimeMultipart;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Email {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Email.class);
+  private final EmailBody body;
+  private final String messageId;
+  private final String subject;
+  private final List<String> from;
+  private final List<String> to;
+  private final List<String> cc;
+  private final OffsetDateTime sentAt;
+  private final OffsetDateTime receivedAt;
+  private final Integer size;
+
+  private Email(
+      EmailBody body,
+      String messageId,
+      String subject,
+      List<String> from,
+      List<String> to,
+      List<String> cc,
+      OffsetDateTime sentAt,
+      OffsetDateTime receivedAt,
+      Integer size) {
+    this.body = body;
+    this.messageId = messageId;
+    this.subject = subject;
+    this.from = from;
+    this.to = to;
+    this.cc = cc;
+    this.sentAt = sentAt;
+    this.receivedAt = receivedAt;
+    this.size = size;
+  }
+
+  public static Email createBodylessEmail(Message message) {
+    try {
+      List<String> from =
+          Arrays.stream(Optional.ofNullable(message.getFrom()).orElse(new Address[0]))
+              .map(Address::toString)
+              .map(address -> address.replaceAll(".*<|>.*", ""))
+              .toList();
+      List<String> to =
+          Arrays.stream(
+                  Optional.ofNullable(message.getRecipients(Message.RecipientType.TO))
+                      .orElse(new Address[0]))
+              .map(Address::toString)
+              .map(address -> address.replaceAll(".*<|>.*", ""))
+              .toList();
+      List<String> cc =
+          Arrays.stream(
+                  Optional.ofNullable(message.getRecipients(Message.RecipientType.CC))
+                      .orElse(new Address[0]))
+              .map(Address::toString)
+              .map(address -> address.replaceAll(".*<|>.*", ""))
+              .toList();
+      OffsetDateTime sentAt =
+          Optional.ofNullable(message.getSentDate())
+              .map(Date::toInstant)
+              .map(instant -> instant.atOffset(ZoneOffset.UTC))
+              .orElse(null);
+      OffsetDateTime receivedAt =
+          Optional.ofNullable(message.getReceivedDate())
+              .map(Date::toInstant)
+              .map(instant -> instant.atOffset(ZoneOffset.UTC))
+              .orElse(null);
+      String messageId = stripMessageId(message.getHeader("Message-ID")[0]);
+      return new Email(
+          null,
+          messageId,
+          message.getSubject(),
+          from,
+          to,
+          cc,
+          sentAt,
+          receivedAt,
+          message.getSize());
+
+    } catch (MessagingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static Email createEmail(Message message) {
```

Comment:
I would recommend to handle this outside of the `Email` model class.

Reply:
ok 😁